exonum/src/node/basic.rs

```diff
@@ -14,32 +14,36 @@
 
 use rand::{self, Rng};
 
-use std::{error::Error, net::SocketAddr};
+use std::net::SocketAddr;
 
 use super::{NodeHandler, NodeRole, RequestData};
+use events::error::LogError;
 use helpers::Height;
-use messages::{Any, Connect, Message, PeersRequest, RawMessage, Status};
+use messages::{Connect, Message, PeersRequest, Protocol, Responses, Service, Status};
 
 impl NodeHandler {
     /// Redirects message to the corresponding `handle_...` function.
-    pub fn handle_message(&mut self, raw: RawMessage) {
-        match Any::from_raw(raw) {
-            Ok(Any::Connect(msg)) => self.handle_connect(msg),
-            Ok(Any::Status(msg)) => self.handle_status(&msg),
-            Ok(Any::Consensus(msg)) => self.handle_consensus(msg),
-            Ok(Any::Request(msg)) => self.handle_request(msg),
-            Ok(Any::Block(msg)) => self.handle_block(&msg),
-            Ok(Any::Transaction(msg)) => self.handle_tx(&msg),
-            Ok(Any::TransactionsBatch(msg)) => self.handle_txs_batch(&msg),
-            Err(err) => {
-                error!("Invalid message received: {:?}", err.description());
+    pub fn handle_message(&mut self, msg: Protocol) {
+        match msg {
+            Protocol::Consensus(msg) => self.handle_consensus(msg),
+            Protocol::Requests(msg) => self.handle_request(msg),
+
+            Protocol::Service(Service::Connect(msg)) => self.handle_connect(msg),
+            Protocol::Service(Service::Status(msg)) => self.handle_status(msg),
+            // ignore tx duplication error,
+            Protocol::Service(Service::RawTransaction(msg)) => drop(self.handle_tx(msg)),
```

Comment:
Why is `drop()` needed now? Did we start returning a `Result` here which must be used? Is it okay to silently ignore the error without handling it or logging it (at least with trace/debug level)? It may cause issues when `handle_tx()` starts reporting not only duplicates, but other errors as well.

Reply:
Before this pr this error was ignored too (inside `handle_tx_verified` as far as i remember), i just moved this code, because of thread-pool verification refactor (now we checks messages instead of transactions).

I like the idea to add debug level. But our contributing.md says that we should avoid using debugs. 

Probably the wole `handle_XX` code should be refactored, for better error handling. But i prefer to move this outside of current refactor scope.